spec/unit/pdk/cli/env_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'spec_helper'
+require 'pdk/cli'
+
+describe 'Running `pdk env`' do
+  let(:command_args) { ['env'] }
+  let(:command_result) { { exit_code: 0 } }
+
+  context 'when it calls env successfully' do
+    after(:each) do
+      expect {
+        PDK::CLI.run(command_args)
+      }.to exit_zero
+    end
+
+    before(:each) do
+      allow(PDK::Util::RubyVersion).to receive(:gem_home).and_return("/opt/puppetlabs/pdk/share/cache/ruby/2.4.0")
+      allow(PDK::Util::RubyVersion).to receive(:gem_path).and_return("/opt/puppetlabs/pdk/private/ruby/2.4.3/lib")
+      allow(PDK::Util::RubyVersion).to receive(:bin_path).and_return("/opt/puppetlabs/pdk/private/ruby/2.4.3/bin")
+      allow(PDK::Util::RubyVersion).to receive(:gem_paths_raw).and_return(["/opt/puppetlabs/pdk/private/ruby/2.4.3/lib"])
+      allow(PDK::Util::Env).to receive(:[]).and_call_original
+      allow(PDK::Util::Env).to receive(:[]).with('PATH').and_return('/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin')
+
+      allow(PDK::CLI::Util).to receive(:puppet_from_opts_or_env)
+        .and_return(ruby_version: '2.4.3', gemset: { puppet: '5.4.0' })
+      allow(PDK::Util::RubyVersion).to receive(:use)
+    end
+
+    context 'and called with no arguments' do
+      it 'sends a "env" screen view to analytics' do
+        expect(analytics).to receive(:screen_view).with(
+          'env',
+          output_format: 'default',
+          ruby_version:  RUBY_VERSION,
+        )
+      end
+
+      it 'outputs export commands for environment variables' do
+        output_regexes = [
+          /export PDK_RESOLVED_PUPPET_VERSION="\d\.\d+\.\d+"/,
+          /export PDK_RESOLVED_RUBY_VERSION="\d\.\d+\.\d+"/,
+          /export GEM_HOME=.*/,
+          /export GEM_PATH=.*/,
+          /export PATH=.*/
+        ]
+
+        output_regexes.each do |regex|
+          expect($stdout).to receive(:puts).with(a_string_matching(regex))
+        end
+      end
+    end
+
+    context 'and called with a puppet version' do
+      let(:command_args) { super() + %w[--puppet-version=6] }
+
+      it 'outputs export commands for environment variables' do
+        output_regexes = [
+          /export PDK_RESOLVED_PUPPET_VERSION="\d\.\d+\.\d+"/,
+          /export PDK_RESOLVED_RUBY_VERSION="\d\.\d+\.\d+"/,
+          /export GEM_HOME=.*/,
+          /export GEM_PATH=.*/,
+          /export PATH=.*/
+        ]
+
+        output_regexes.each do |regex|
+          expect($stdout).to receive(:puts).with(a_string_matching(regex))
+        end
+      end
+
+      # context 'with a conflicting environment variable' do
```

Comment:
@scotje I believe you mentioned adding a test for the case where there's an env var for the puppet version that conflicts with the option passed to the command. I think this block would cover that, except since `puppet_from_opts_or_env` is mocked, this isn't meaningful as-is. Should that `puppet_from_opts_or_env` method be handled in some other way? Or is there another way to test for an env var vs. option conflict?